src/S3/MultipartUploadingTrait.php

```diff
@@ -49,12 +51,24 @@ public static function getStateFromService(
         return $state;
     }
 
+    protected function createProgressThresholds($totalSize)
+    {
+        $this->progressThresholds = [];
+        for ($i=1;$i<=8;$i++) {
+            $this->progressThresholds []= round($totalSize*($i/8));
+        }
+        $this->getState()->setProgressThresholds($this->progressThresholds);
```

Comment:
I think it would be better to set this in the constructor of `UploadState`.  That includes moving this logic to that class (`setProgressThresholds`).  Generally speaking, you want to set the initial values of your class members in the constructor of a given class.